### ISSUE TITLE
Concurrent queue locking up

### ISSUE THREAD

**Issue by michaeldwilliams** [marmelroy/PhoneNumberKit#528]
We are experiencing an issue running a value through your parser where the concurrent queue (created in `RegexManager`) is locking up when attempting to retrieve a cached value from the `regularExpressionPool`. 

On our end, we are retrieving an array of objects from our backend and as we create local objects from that, we are running each phone number from the backend objects through parser. There is nothing strange or particular about the numbers it locks up on. Just "1234567890".

```
    func foo(id: [String]) async throws -> [Foo] {
        if id.isEmpty {
            return []
        }

        let response = ...
        response.errors.forEach { log.error("\($0)") }
        let foos = response.array.compactMap {
            try? Foo(responseItem: $0)
        }
        return foos
    }

  ...

    init(responseItem: Response) throws {
        do {
            if let daytimePhone = responseItem.locationContact.daytimePhone {
                let combinedPhoneComponents = daytimePhone.areaCode + daytimePhone.phoneNumber
                let phoneNumber = try PhoneNumberUtilities.shared.phoneNumberDigits(from: combinedPhoneComponents)
                self.phoneNumber = phoneNumber.string
            } else {
                self.phoneNumber = nil
            }
        } catch {
            log.error("Error: \(error)")
            self.phoneNumber = nil
        }
   }

   ...

    func phoneNumberDigits(from string: String) throws -> Int {
        let digitString = string
            .components(separatedBy: CharacterSet.decimalDigits.inverted)
            .joined()
        return Int(try pnk.parse(digitString).nationalNumber)
    }
```

When I remove the parse call and just send the string through with no validation via PNK, we have no issues. Also, after pulling down the project, I removed the `.concurrent` attribute and that seems to fix the issue as well.

**Comment by bguidolim** [comment]
Seems it was made like this to fix #193, however, it is working with the single purpose of making the access to the cache thread-safe, but it's not properly configured.

It's a quick fix, I'll make a PR soon.

**Comment by bguidolim** [comment]
@michaeldwilliams could you please check if my PR fixes the issue?
